SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/ReferenceCountedLockingPolicy.cpp

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "LoadBalancing/ReferenceCountedLockingPolicy.h"
+#include "GameFramework/Actor.h"
+
+#include "EngineClasses/SpatialNetDriver.h"
+#include "EngineClasses/SpatialPackageMapClient.h"
+#include "Interop/SpatialStaticComponentView.h"
+#include "Schema/AuthorityIntent.h"
+
+DEFINE_LOG_CATEGORY(LogReferenceCountedLockingPolicy);
+
+ActorLockToken UReferenceCountedLockingPolicy::AcquireLock(const AActor* Actor)
+{
+	check(Actor != nullptr);
+	check(CanAcquireLock(Actor))
+	++ActorToReferenceCount.FindOrAdd(Actor);
+	TokenToNameAndActor.Emplace(NextToken, LockNameAndActor{ FString(), Actor });
+	UE_LOG(LogReferenceCountedLockingPolicy, Log, TEXT("Acquiring migration lock. Actor: %s. Locks held: %d."), *Actor->GetName(), *ActorToReferenceCount.Find(Actor));
+	return NextToken++;
+}
+
+ActorLockToken UReferenceCountedLockingPolicy::AcquireLock(const AActor* Actor, FString DebugString)
+{
+	check(Actor != nullptr);
+	check(CanAcquireLock(Actor))
+	++ActorToReferenceCount.FindOrAdd(Actor);
+	UE_LOG(LogReferenceCountedLockingPolicy, Log, TEXT("Acquiring migration lock. "
+		"Actor: %s. Lock name: %s. Token %d: Locks held: %d."), *Actor->GetName(), *DebugString, NextToken, *ActorToReferenceCount.Find(Actor));
+	TokenToNameAndActor.Emplace(NextToken, LockNameAndActor{ MoveTemp(DebugString), Actor });
+	return NextToken++;
+}
+
+bool UReferenceCountedLockingPolicy::CanAcquireLock(const AActor* Actor) const
```

Comment:
Since every other function has `check(Actor != nullptr);`, I think it should be added here as well.